cardano-cli/src/Cardano/CLI/EraBased/Run/StakeAddress.hs

```diff
@@ -325,3 +328,31 @@ runStakeAddressDeregistrationCertificateCmd sbe stakeVerifier mDeposit oFp = do
  where
   deregCertDesc :: TextEnvelopeDescr
   deregCertDesc = "Stake Address Deregistration Certificate"
+
+runStakeAddressRegistrationAndDelegationCertificateCmd
+  :: ()
+  => ConwayEraOnwards era
+  -> StakeIdentifier
+  -> VerificationKeyOrHashOrFile StakePoolKey
+  -- ^ Delegatee stake pool verification key or verification key file or
+  -> Lovelace
+  -> File () Out
+  -> ExceptT StakeAddressCmdError IO ()
+runStakeAddressRegistrationAndDelegationCertificateCmd w stakeVerifier poolVKeyOrHashOrFile deposit outFp = 
+  conwayEraOnwardsConstraints w $ do
+    StakePoolKeyHash poolStakeVKeyHash <-
+      modifyError StakeAddressCmdReadKeyFileError $
+        readVerificationKeyOrHashOrFile AsStakePoolKey poolVKeyOrHashOrFile
+  
+    stakeCred <-
+      getStakeCredentialFromIdentifier stakeVerifier
+        & firstExceptT StakeAddressCmdStakeCredentialError 
+  
+    let deleg = L.DelegStake poolStakeVKeyHash
+
+    let certificate = makeStakeAddressAndDRepDelegationCertificate w stakeCred deleg deposit
```

Comment:
Question, @smelc 

I'm using  `makeStakeAddressAndDRepDelegationCertificate`  from API 

```
makeStakeAddressAndDRepDelegationCertificate
  :: ()
  => ConwayEraOnwards era
  -> StakeCredential
  -> Ledger.Delegatee (EraCrypto (ShelleyLedgerEra era))
  -> L.Coin
  -> Certificate era
makeStakeAddressAndDRepDelegationCertificate w cred delegatee deposit =
  conwayEraOnwardsConstraints w $
    ConwayCertificate w $
      Ledger.mkRegDepositDelegTxCert (toShelleyStakeCredential cred) delegatee deposit
```

The AndDRepDelegation may cause some noise, but I think this function should be renamed to something more general because `Ledger.Delegatee` can actually be: 

```
data Delegatee c
  = DelegStake !(KeyHash 'StakePool c)
  | DelegVote !(DRep c)
  | DelegStakeVote !(KeyHash 'StakePool c) !(DRep c)
  deriving (Show, Generic, Eq, Ord)
  ```
thus I use:   
```  
let deleg = L.DelegStake poolStakeVKeyHash
```

Reply:
@CarlosLopezDeLara> Yes, good choice using `makeStakeAddressAndDRepDelegationCertificate` :+1: 

And yes I think it makes sense to rename it. The ledger's name `mkRegDepositDelegTxCert` is not so good IMHO, so I'd go with `makeStakeAddressDelegationCertificate` (omitting the DRep mention, as you proposed).